src/main/java/org/wso2/carbon/connector/pcml/AS400Initialize.java

```diff
@@ -0,0 +1,352 @@
+
+            // Authenticating user if user ID and password are set.
+            if (!userID.isEmpty() && !password.isEmpty()) {
+                as400.authenticate(userID, password);
+                log.auditLog("Authentication success...");
+            }
+
+        } catch (AS400SecurityException as400SecurityException) {
+            String errorMessage = "Security or authorization error occurred: ";
+            AS400Utils.setExceptionToPayload(errorMessage, as400SecurityException, "100", messageContext);
+            handleException(errorMessage + as400SecurityException.getMessage(), as400SecurityException, messageContext);
+        } catch (IOException ioException) {
+            String errorMessage = "Error occurs when communicating to the AS400 server: ";
+            AS400Utils.setExceptionToPayload(errorMessage, ioException, "101", messageContext);
+            handleException(errorMessage + ioException.getMessage(), ioException, messageContext);
+        } catch (PropertyVetoException guiDisableException) {
+            String errorMessage = "Unable to disable GUI mode of JTOpen library for authenticating: ";
+            AS400Utils.setExceptionToPayload(errorMessage, guiDisableException, "102", messageContext);
+            handleException(errorMessage + guiDisableException.getMessage(), guiDisableException, messageContext);
+        } catch (ConnectionPoolException conPoolException) {
+            String errorMessage = "Error occurred while getting a connection from the pool: ";
+            AS400Utils.setExceptionToPayload(errorMessage, conPoolException, "103", messageContext);
+            handleException(errorMessage + conPoolException.getMessage(), conPoolException, messageContext);
+        } catch (Exception exception) {
+            String errorMessage = "Exception occurred while initializing the AS400 instance: ";
+            AS400Utils.setExceptionToPayload(errorMessage, exception, "199", messageContext);
+            handleException(errorMessage + exception.getMessage(), exception, messageContext);
+        } finally {
+            // Adding the as400 object to message context.
+            messageContext.setProperty(AS400Constants.AS400_INSTANCE, as400);
+        }
+    }
+
+    /**
+     * Gets a connection pool from {@link #as400ConnectionPoolMap} with a given pool name. If a connection pool does not
+     * exists, then a new pool is created and stores in the {@link #as400ConnectionPoolMap} and then returned.
+     *
+     * @param messageContext The message context.
+     * @param log            The logger object.
+     * @return The connection pool.
+     */
+    private AS400ConnectionPool getConnectionPool(MessageContext messageContext, SynapseLog log) {
+        AS400ConnectionPool as400ConnectionPool = null;
+        try {
+            Object poolNameParameter = getParameter(messageContext, AS400Constants.AS400_CONNECTION_POOL_NAME);
+            if (null != poolNameParameter) {
+
```

Comment:
Fixed in https://github.com/wso2-extensions/esb-connector-pcml/pull/3/commits/22986634427adb228f51d47163e25fa13cbc10bc